sdk/communication/azure-communication-chat/src/main/java/com/azure/android/communication/chat/ChatAsyncClient.java

```diff
@@ -257,6 +262,62 @@ public ChatThreadAsyncClient getChatThreadClient(String chatThreadId) {
         return this.chatClient.deleteChatThreadWithResponseAsync(chatThreadId, context);
     }
 
+    /**
+     * Receive real-time messages and notifications.
+     */
+    public void startRealtimeNotifications() {
+        if (this.signalingClient == null) {
+            throw new Error("Signaling client not initialized");
+        }
+
+        if (this.isRealtimeNotificationsStarted) {
+            return;
+        }
+
+        this.isRealtimeNotificationsStarted = true;
+        this.signalingClient.start();
+    }
+
+    /**
+     * Stop receiving real-time messages and notifications.
+     */
+    public void stopRealtimeNotifications() {
+        if (this.signalingClient == null) {
+            throw new Error("Signaling client not initialized");
+        }
+
+        this.isRealtimeNotificationsStarted = false;
+        this.signalingClient.stop();
+    }
+
+    /**
+     * Listen to a chat event.
+     */
+    public void on(String chatEventId, String listenerId, RealTimeNotificationCallback listener) {
+        if (this.signalingClient == null) {
+            throw new Error("Realtime notification parameters (context, userToken) are not set");
```

Comment:
For all exception throwing, we should direct it through `logger.*` API. Also I would use `IllegalStateException` instead of `Error` (please apply this comment for other places where we do similar throwing)